src/features/supervisor-app/hooks/supervisor-app.ts

```diff
@@ -221,31 +236,41 @@ async function setSupervisorReleaseResource(
 		},
 	});
 
-	return Promise.all(
+	await Promise.all(
 		_.map(devicesByDeviceTypeArch, async (affectedDevices, deviceTypeArch) => {
-			const affectedDeviceIds = affectedDevices.map((d) => d.id);
+			const affectedDeviceIds = affectedDevices.map((d) => d.id as number);
 
 			const [supervisorRelease] = await getSupervisorReleaseResource(
 				api,
 				supervisorVersion,
 				deviceTypeArch,
 			);
 
-			if (supervisorRelease == null) {
-				return;
+			if (supervisorRelease?.id) {
+				await setSupervisorRelease(
+					rootApi,
+					affectedDeviceIds,
+					supervisorRelease.id,
+				);
 			}
-
-			await rootApi.patch({
-				resource: 'device',
-				options: {
-					$filter: {
-						id: { $in: affectedDeviceIds },
-					},
-				},
-				body: {
-					should_be_managed_by__release: supervisorRelease.id,
-				},
-			});
 		}),
 	);
 }
+
+function setSupervisorRelease(
```

Comment:
This is a trivial function and only used from one place -- does it have to be separate?

Reply:
The next PR will make it more clear why we are refactoring this into a separate function.  The purpose is to reduce the cyclomatic complexity of the calling method, which in the next PR will start to grow otherwise.